src/solvers/flattening/boolbv_add_sub.cpp

```diff
@@ -33,17 +37,20 @@ bvt boolbvt::convert_add_sub(const exprt &expr)
 
   const exprt::operandst &operands=expr.operands();
 
-  if(operands.empty())
-    throw "operator "+expr.id_string()+" takes at least one operand";
+  DATA_INVARIANT(
+    !operands.empty(),
+    "operator " + expr.id_string() + " takes at least one operand");
 
   const exprt &op0=expr.op0();
   DATA_INVARIANT(
     op0.type() == type, "add/sub with mixed types:\n" + expr.pretty());
 
   bvt bv=convert_bv(op0);
 
-  if(bv.size()!=width)
-    throw "convert_add_sub: unexpected operand 0 width";
+  INVARIANT(
```

Comment:
I'd say a `CHECK_RETURN` would do.

Reply:
But really: we are doing this all the time. Couldn't we just make `convert_bv` take a second argument `expected_width` and do that check itself? This will also take the error reporting one step closer to where the error occurred, helping to fix root causes, not symptoms.